astropy/tests/runner.py

```diff
@@ -207,33 +292,112 @@ def run_tests(self, package=None, test_path=None, args=None, plugins=None,
 
             print("Checking for unclosed files")
 
+        return []
+
+    @keyword(0)
+    def parallel(self, parallel, kwargs):
+        """
+        parallel : int, optional
+            When provided, run the tests in parallel on the specified
+            number of CPUs.  If parallel is negative, it will use the all
+            the cores on the machine.  Requires the `pytest-xdist` plugin.
+
+        """
         if parallel != 0:
-            try:
-                import xdist  # pylint: disable=W0611
-            except ImportError:
-                raise ImportError(
-                    'Parallel testing requires the pytest-xdist plugin '
-                    'https://pypi.python.org/pypi/pytest-xdist')
+            return ['-n', six.text_type(parallel)]
 
-            try:
-                parallel = int(parallel)
-            except ValueError:
-                raise ValueError(
-                    "parallel must be an int, got {0}".format(parallel))
+        return []
 
-            if parallel < 0:
-                parallel = multiprocessing.cpu_count()
-            all_args.extend(['-n', six.text_type(parallel)])
+    @keyword
+    def docs_path(self, docs_path, kwargs):
+        """
+        docs_path : str, optional
+            The path to the documentation .rst files.
 
+        """
+        if docs_path is not None and not kwargs['skip_docs']:
+            if kwargs['package'] is not None:
+                docs_path = os.path.join(
+                    docs_path, kwargs['package'].replace('.', os.path.sep))
+            if not os.path.exists(docs_path):
+                warnings.warn(
+                    "Can not test .rst docs, since docs path "
+                    "({0}) does not exist.".format(docs_path))
+                docs_path = None
+        if docs_path and not kwargs['skip_docs'] and not kwargs['test_path']:
+            return [docs_path, '--doctest-rst']
+
+        return []
+
+    @keyword
+    def skip_docs(self, skip_docs, kwargs):
+        """
+        skip_docs : bool, optional
+            When `True`, skips running the doctests in the .rst files.
+
+        """
+        # Skip docs is a bool used by docs_path only.
+        return []
+
+    @keyword
+    def repeat(self, repeat, kwargs):
+        """
+        repeat : int, optional
+            If set, specifies how many times each test should be run. This is
+            useful for diagnosing sporadic failures.
+
+        """
         if repeat:
-            all_args.append('--repeat={0}'.format(repeat))
+            return ['--repeat={0}'.format(repeat)]
+
+        return []
+
+    def run_tests(self, **kwargs):
+        """
+        Run the tests
+
+        Parameters
+        ----------
+        {keywords}
+        See Also
+        --------
+        pytest.main : py.test function wrapped by `run_tests`.
+
+        """
+
+        # Don't import pytest until it's actually needed to run the tests
+        from .helper import pytest
```

Comment:
Good question - I'm not sure why it had to be this way. Maybe just try and put it top-level?